### ISSUE TITLE
Search parser missinterprets / in search terms

### ISSUE THREAD

**Issue by aw-bib** [inveniosoftware/invenio#2805]
Given a query with an even amount of / in the search terms, like

    "10.1016/j.ijms.2014.02.006" or "10.1016/j.ijms.2014.02.006"

In this case, even if the term as such matches, invenio returns no results. If the number of / is odd, however, the it returns results, e.g.

    "10.1016/j.ijms.2014.02.006" or "10.1016/j.ijms.2014.02.006"  "10.1016/j.ijms.2014.02.006"

or

    "10.1016/j.ijms.2014.02.006"

It looks like invenio's query parser is treating even numbers of / as regexp pattern search even though they are enclosed in quotes like above.

Breaks on invenio up to at least 1.1.4. The above example is for cds.cern.ch (ie. the record with this id exists there). This behaviour was first observed at JuSER (http://juser.fz-juelich.de) with the query

    id:"10.7566/JPSJS.82SA.SA016" or id:"10.1017/S1461145712000697"

running invenio 1.1.4 (tag release 1.1.4)

**Comment by kaplun** [comment]
This bug should not exist on next as it is related to the native legacy search query parser.

**Comment by aw-bib** [comment]
@kaplun I'm not sure bout the v1.3 milestone. a fix should get backported to 1.1 tree, please.

**Comment by kaplun** [comment]
v1.3 means it will probably not go into the next 1.2.0 major release, since it due 3 months ago, and we don't necessarily have the resources right now to focus on this specific bug. Volunteers are always welcomed :-)

**Comment by kaplun** [comment]
(any available fix will likely be made available first on maint-1.1)

**Comment by aw-bib** [comment]
@kaplung understood. I admit to fix it myself I'd need a bit of a pointer where to look. The search engine internals are complex.

**Comment by kaplun** [comment]
Indeed! That's why a quick fix can't be promised. Somehow the bug lays in `invenio.search_engine.create_basic_search_units()` 
https://github.com/inveniosoftware/invenio/blob/maint-1.1/modules/websearch/lib/search_engine.py#L680-L837

**Comment by aw-bib** [comment]
Just found by @cfrick-fzj

It also seesm that a query like 

    id:"10.1175/BAMS-D-11-00123.1" or id:"10.1016/S1389-1286(00)00162-6"

goes entrily a miss in 1.1 (tested on 1.1.5 at http://juser.fz-juelich.de, should give two hits, ie both DOIs are there.)

    Search stage 1: basic search units are: [['+', '10.1175/BAMS-D-11-00123.1 or id:10.1016/S1389-1286(00)00162-6', 'id', 'a'], ['|', '10.1016/S1389-1286(00)00162-6', 'id', 'a']]

These types of queries are quite important for record curation, where one needs a reliable way for exact hits.

**Comment by jirikuncar** [comment]
This has been solved in `invenio-query-parser` (*Invenio v2.1.dev+*):

```
In [2]: s = SearchEngine('"10.1016/j.ijms.2014.02.006" or "10.1016/j.ijms.2014.02.006"')

In [3]: s.query
Out[3]: OrOp(ValueQuery(DoubleQuotedValue('10.1016/j.ijms.2014.02.006')), ValueQuery(DoubleQuotedValue('10.1016/j.ijms.2014.02.006')))

In [4]: s = SearchEngine('10.1016/j.ijms.2014.02.006 or 10.1016/j.ijms.2014.02.006')

In [5]: s.query
Out[5]: OrOp(ValueQuery(Value('10.1016/j.ijms.2014.02.006')), ValueQuery(Value('10.1016/j.ijms.2014.02.006')))
```

**Comment by aw-bib** [comment]
Is there a backport of this fix to maint(-1.1)? This would indeed be quite important at least to us.

**Comment by jirikuncar** [comment]
@aw-bib it's not easily backportable. Maybe one can use `invenio-query-parser` to generate only basic search unit.
